code/modules/mob/living/simple_animal/hostile/terror_spiders/objects.dm

```diff
@@ -0,0 +1,353 @@
+	var/spider_growinstantly = 0
+	var/spider_queen_declared_war = 0 // set to 1 by procs
+	var/faction = list("terrorspiders")
+	var/spider_myqueen = null
+	var/master_commander = null
+	var/spiderling_type = null
+	var/spiderling_number = 1
+	var/spiderling_ventcrawl = 1
+	var/ai_playercontrol_allowingeneral = 1
+	var/list/enemies = list()
+
+
+/obj/effect/spider/terror_eggcluster/New()
+	pixel_x = rand(3,-3)
+	pixel_y = rand(3,-3)
+	processing_objects.Add(src)
+
+
+/obj/effect/spider/terror_eggcluster/process()
+	amount_grown += rand(0,2)
+	if(amount_grown >= 100)
+		var/num = spiderling_number
+		for(var/i=0, i<num, i++)
+			var/obj/effect/spider/terror_spiderling/S = new /obj/effect/spider/terror_spiderling(get_turf(src))
+			if (spiderling_type)
+				S.grow_as = spiderling_type
+			if (spiderling_ventcrawl)
+				S.use_vents = spiderling_ventcrawl
+			if (S.grow_as == "/mob/living/simple_animal/hostile/poison/terror_spider/queen")
+				S.name = "queen spiderling"
+			else if (S.grow_as == "/mob/living/simple_animal/hostile/poison/terror_spider/red")
+				S.name = "red spiderling"
+			else if (S.grow_as == "/mob/living/simple_animal/hostile/poison/terror_spider/black")
+				S.name = "black spiderling"
+			else if (S.grow_as == "/mob/living/simple_animal/hostile/poison/terror_spider/green")
+				S.name = "green spiderling"
+			else if (S.grow_as == "/mob/living/simple_animal/hostile/poison/terror_spider/purple")
+				S.name = "purple spiderling"
+			else if (S.grow_as == "/mob/living/simple_animal/hostile/poison/terror_spider/white")
+				S.name = "white spiderling"
+			S.faction = faction
+			S.spider_myqueen = spider_myqueen
+			S.master_commander = master_commander
+			S.ai_playercontrol_allowingeneral = ai_playercontrol_allowingeneral
+			S.enemies = enemies
+			if (spider_queen_declared_war)
+				S.spider_queen_declared_war = 1
+			if (spider_growinstantly)
+				S.amount_grown = 250
+		var/rnum = 5 - spiderling_number
+		for(var/i=0, i<rnum, i++)
+			var/obj/effect/spider/terror_spiderling/S = new /obj/effect/spider/terror_spiderling(get_turf(src))
+			S.stillborn = 1
+			// the idea is that every set of eggs always spawn 5 spiderlings, but most are not going to grow up, just some do.
+		qdel(src)
+
+
+/obj/effect/spider/terror_spiderling
+	name = "spiderling"
+	desc = "A fast-moving tiny spider, prone to making aggressive hissing sounds. Hope it doesn't grow up."
+	icon_state = "spiderling"
+	anchored = 0
+	layer = 2.75
+	health = 3
+	var/amount_grown = 0
+	var/grow_as = null
+	var/spider_queen_declared_war = 0 // set to 1 by inheritance
+	var/stillborn = 0
+	var/obj/machinery/atmospherics/unary/vent_pump/entry_vent
+	var/travelling_in_vent = 0
+	var/faction = list("terrorspiders")
+	var/spider_myqueen = null
+	var/master_commander = null
+	var/use_vents = 1
+	var/ai_playercontrol_allowingeneral = 1
+	var/list/enemies = list()
+	var/stat = 0 // necessary, because without it spiderlings moving into borg rechargers create a runtime error.
```

Comment:
the solution here is making borg rechargers not runtime if bumped by an object
not adding a pointless stat var to a path